loggingservice.go

```diff
@@ -26,35 +26,84 @@ type LoggerInterface interface {
 }
 
 type StackdriverClient struct {
-	client *logging.Client
+	loggingClient *LoggingClient
+}
+
+type LoggingClient struct {
+	sync.Mutex
+	client      *logging.Client
+	initialized bool
+}
+
+var globalLoggingClient LoggingClient
+
+func (lc *LoggingClient) Logger(logID string, opts ...logging.LoggerOption) *logging.Logger {
+	return lc.client.Logger(logID, opts...)
+}
+
+func (lc *LoggingClient) SetUpOnError() {
+	lc.client.OnError = func(e error) {
+		fmt.Fprintf(os.Stderr, "logging error: %v", e)
+	}
+}
+
+func (lc *LoggingClient) Ping(ctx context.Context) error {
+	return lc.client.Ping(ctx)
```

Comment:
I also added checks in the rest of the methods; but since they don't return errors i panic. This feels not great.... but wasn't sure we'd want to guard in one place but not the rest?

Reply:
What you did is fine! As long as you prevent a panic on nil, I'm happy!